core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java

```diff
@@ -174,11 +175,21 @@ public void setUp() throws Exception {
       File file = (File) invocationOnMock.getArguments()[0];
       return Utils.tempFileWith(file);
     });
-
+    resetDependency();
     when(taskContext.taskMetrics()).thenReturn(taskMetrics);
+    when(taskContext.taskMemoryManager()).thenReturn(taskMemoryManager);
+  }
+
+  private void resetDependency() {
```

Comment:
in tests, we don't need to modify the conf directly. We can just add a new parameter here: `resetDependency(boolean rowBasedCheckSumEnabled)`

Reply:
done